File: app/src/main/java/com/ddu/ui/fragment/study/ui/ImageFragment.java
package com.ddu.ui.fragment.study.ui;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.view.View;
import android.widget.ImageView;

import com.ddu.R;
import com.ddu.icore.ui.fragment.DefaultFragment;
import com.ddu.icore.util.sys.ViewUtils;

import java.lang.reflect.Field;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;

/**
 * Created by lhz on 16/5/6.
 */
public class ImageFragment extends DefaultFragment {

    private Unbinder unbinder;

    private boolean isBigModel = true;

    @BindView(R.id.iv_common)
    ImageView mIvCommon;

    @NonNull
    public static ImageFragment newInstance(String taskId) {
        Bundle arguments = new Bundle();
        arguments.putString(ARGUMENT_TASK_ID, taskId);
        ImageFragment fragment = new ImageFragment();
        fragment.setArguments(arguments);
        return fragment;
    }

    @Override
    public void initData(Bundle savedInstanceState) {

    }

    @Override
    public int getLayoutId() {
        return R.layout.fragment_ui_image;
    }

    @Override
    public void initView() {
        unbinder = ButterKnife.bind(this, mView);

        setDefaultTitle("Image");
        setRightText("小图", new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mTitleBar.getRightText().setText(isBigModel ? "小图" : "大图");
                setImg(isBigModel);
                isBigModel = !isBigModel;
            }
        });
    }

    private void setImg(boolean isBigModel) {
        int resId = isBigModel ? R.drawable.home_icon_glod_card : R.drawable.guide_hand;
        for (int i = 1; i < 10; i++) {
            ImageView imageView = ViewUtils.findViewById(mView, getResId("iv" + i, R.id.class));
            imageView.setImageResource(resId);
        }
        mIvCommon.setImageResource(resId);
    }

    public static int getResId(String variableName, Class<?> c) {
        try {
            Field idField = c.getDeclaredField(variableName);
            return idField.getInt(idField);
        } catch (Exception e) {
            e.printStackTrace();
            return -1;
        }
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        unbinder.unbind();
    }

}

File: app/src/main/java/com/ddu/ui/fragment/study/ui/ImageFragment.kt
package com.ddu.ui.fragment.study.ui

import android.os.Bundle
import android.view.View
import android.widget.ImageView
import com.ddu.R
import com.ddu.icore.ui.fragment.DefaultFragment
import com.ddu.icore.util.sys.ViewUtils
import com.iannotation.ContentType
import kotlinx.android.synthetic.main.fragment_ui_image.*

/**
 * Created by lhz on 16/5/6.
 */
@ContentType("UI")
class ImageFragment : DefaultFragment() {

    private var isBigModel = true

    override fun getLayoutId(): Int {
        return R.layout.fragment_ui_image
    }

    override fun initView() {

        setDefaultTitle("Image")
        setRightText("",object : View.OnClickListener{
            override fun onClick(v: View?) {
                titleBar?.rightText?.text = if (isBigModel) "小图" else "大图"
                setImg(isBigModel)
                isBigModel = !isBigModel
            }
        })
    }

    private fun setImg(isBigModel: Boolean) {
        val resId = if (isBigModel) R.drawable.guide_hand else R.drawable.guide_hand
        for (i in 1..9) {
            val imageView = ViewUtils.findViewById<ImageView>(mView, getResId("iv" + i, R.id::class.java))
            imageView.setImageResource(resId)
        }
        iv_common.setImageResource(resId)
    }

    companion object {

        fun newInstance(taskId: String): ImageFragment {
            val arguments = Bundle()
            arguments.putString(DefaultFragment.ARGUMENT_TASK_ID, taskId)
            val fragment = ImageFragment()
            fragment.arguments = arguments
            return fragment
        }

        fun getResId(variableName: String, c: Class<*>): Int {
            try {
                val idField = c.getDeclaredField(variableName)
                return idField.getInt(idField)
            } catch (e: Exception) {
                e.printStackTrace()
                return -1
            }

        }
    }

}
